### PR TITLE
97035:changes related to user-learner-progress

### PR DESCRIPTION
**Pull request by ravisaurav-tarento**
Hi @karthik-tarento please review the PR, and suggest changes, if any.

### CODE REVIEW

File: src/main/java/org/sunbird/progress/service/MandatoryContentServiceImpl.java [hunk 1]
```diff
@@ -178,15 +187,24 @@ public Map<String, Object> getUserProgress(SunbirdApiRequest requestBody, String
...
-			List<String> enrolledUserId = userEnrolmentList.stream()
-					.map(obj -> (String) obj.get(Constants.USER_ID_CONSTANT)).collect(Collectors.toList());
 			List<String> userFields = new ArrayList<>(Arrays.asList(Constants.USER_ID_CONSTANT, Constants.FIRSTNAME, Constants.PROFILE_DETAILS_PRIMARY_EMAIL, Constants.CHANNEL,
 					Constants.PROFILE_DETAILS_DESIGNATION, Constants.PROFILE_DETAILS_DESIGNATION_OTHER));
-			Map<String, Object> userMap = userUtilService.getUsersDataFromUserIds(enrolledUserId, userFields,
+			Map<String, Object> userMap = userUtilService.getUsersDataFromUserIds(enrollmentIdList, userFields,
 					authUserToken);
 
 			Map<String, Integer> courseLeafCount = new HashMap<>();
+			if (!courseLeafCount.containsKey(requestData.getCourseId())) {
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/progress/service/MandatoryContentServiceImpl.java]: This should be outside of the getBatchList for loop.

**Review by karthik-tarento — CHANGES_REQUESTED** [review]

**Comment by ravisaurav-tarento** [comment]
Hi @karthik-tarento, the request changes have been made. Kindly review it again.

**Review by karthik-tarento — APPROVED** [review]

**Comment by ravisaurav-tarento** [comment]
Additional suggested change also made.

File: src/main/java/org/sunbird/progress/service/MandatoryContentServiceImpl.java (hunk 1 above)
Comment (ravisaurav-tarento) [reply src/main/java/org/sunbird/progress/service/MandatoryContentServiceImpl.java]: Changes made as suggested @karthik-tarento

**Pull request merged**